admin/acceptance.py

```diff
@@ -220,38 +227,47 @@ def stop_nodes(self):
                 print "Failed to destroy %s: %s" % (node.name, e)
 
 
-def rackspace_runner(config, **kwargs):
+def runner_for_cloudprovisioner(provisioner_name):
     """
-    Run the tests against rackspace nodes.
+    Dynamically generate and return a factory function which will be used to
+    build a ``LibcloudRunner`` instance.
+
+    :param unicode provisioner_name: The name of the cloud provisioner for
+        which to build a runner factory function.
+    :returns: a factory function for the given ``provisioner_name``.
+    :raises: ``UsageError`` unless the configuration includes a provisioner
+        specific stanza. All cloud provisioners are assumed to require some
+        specific configuration.
     """
-    from flocker.provision import rackspace_provisioner
-    try:
-        rackspace_config = config['rackspace']
-    except KeyError:
-        raise UsageError("Must provided 'rackspace' config stanza.")
-
-    provisioner = rackspace_provisioner(**rackspace_config)
-    return LibcloudRunner(config=config, provisioner=provisioner, **kwargs)
-
+    def runner_factory(config, **kwargs):
+        """
+        :param dict config: The complete configuration.
+        :param kwargs: Extra keyword arguments which will be supplied to
+            ``LibcloudRunner`` initializer.
+        :returns: A ``LibcloudRunner`` instance.
+        """
+        try:
+            provisioner_config = config[provisioner_name]
+        except KeyError:
+            raise UsageError(
+                "Configuration file must include a {!r} config stanza.".format(
+                    provisioner_name)
+            )
 
-def aws_runner(config, **kwargs):
-    """
-    Run the tests against aws nodes.
-    """
-    from flocker.provision import aws_provisioner
-    try:
-        aws_config = config['aws']
-    except KeyError:
-        raise UsageError("Must provided 'aws' config stanza.")
+        provisioner_factory = namedAny(
+            'flocker.provision.{}_provisioner'.format(
+                provisioner_name))
```

Comment:
Would it make sense for `flocker.provision` to provide something where we could look up the provisioners by name, rather than using `namedAny`. (Probably just a dictionary somewhere).